nixops/resources/__init__.py

```diff
@@ -42,6 +42,17 @@ def get_resource_type(cls):
 
     def __init__(self, name: str, config: ResourceEval):
         config_type = self.__annotations__.get("config", ResourceOptions)
+
+        if isinstance(config_type, str):
+            if config_type == "ResourceOptions":
+                raise TypeError(
+                    f'{self.__class__} is missing a "config" attribute, for example: `config: nixops.resources.ResourceOptions`, see https://nixops.readthedocs.io/en/latest/plugins/authoring.html'
```

Comment:
Instead of "for example," maybe we should link to an anchor on the readthedocs page that explains the existence/necessity/usage of `config: nixops.resources.ResourceOptions`. Then this could become:

```suggestion
                    f'{self.__class__} is missing a "config" attribute, see: https://nixops.readthedocs.io/en/latest/plugins/authoring.html#some-anchor-with-info'
```